encoding/src/keys.rs

```diff
@@ -6,10 +6,11 @@
 //
 // author: Ivaylo Petrov <ivajloip@gmail.com>
 use super::securityhelpers::generic_array::{typenum::U16, GenericArray};
+use serde::{Deserialize, Serialize};
 
 /// AES128 represents 128 bit AES key.
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
-#[derive(Debug, Default, PartialEq, Eq, Clone, Copy)]
+#[derive(Debug, Default, PartialEq, Eq, Clone, Copy, Serialize, Deserialize)]
```

Comment:
I might be misunderstanding the intend here, but wouldn't that require serde to build? I think this is a great default, but I am curious how confident can we be that this is always what people will want? Maybe we can provide a new feature and two variants - if the feature is enabled or if it's not? That way users can disable it if they so desire (I can imagine for example gateway that partially parses data and logs something or device that uses an unusual way to store the session might not want this).

Reply:
It could put it behind a feature. Alternatively we could let the user impl it themselves but that is also cumbersome.

I was planning on using Postcard for Serde to get the SessionData into bytes that can be saved to flash but it uses varint so it is variable length. Maybe we should skip serde alltogether and just use `core::slice::raw_from_parts`. Usually the user doesn't really need the SessionData into anything else than bytes. They could impl serde themselves if they need to.

Reply:
I think making serde an optional feature is fine. You can still use fixed length records with postcard, as long as you take into account the max size and fill the rest with zeros. Or, you can use a different encoding that doesn't use varint.

core::slice::raw_from_parts will be compiler and target dependent, so I don't think that's something users should rely on.